server/test/com/cloud/template/TemplateManagerImplTest.java

```diff
@@ -18,20 +18,369 @@
+import org.springframework.context.annotation.FilterType;
+import org.springframework.core.type.classreading.MetadataReader;
+import org.springframework.core.type.classreading.MetadataReaderFactory;
+import org.springframework.core.type.filter.TypeFilter;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import javax.inject.Inject;
+import java.io.IOException;
 
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(loader = AnnotationConfigContextLoader.class)
 public class TemplateManagerImplTest {
 
-    TemplateManagerImpl tmgr = new TemplateManagerImpl();
+    @Inject
+    TemplateManagerImpl templateManager = new TemplateManagerImpl();
+
+    @Inject
+    DataStoreManager dataStoreManager;
+
+    @Inject
+    VMTemplateDao vmTemplateDao;
+
+    @Inject
+    VMTemplatePoolDao vmTemplatePoolDao;
+
+    @Inject
+    TemplateDataStoreDao templateDataStoreDao;
+
+    @Inject
+    StoragePoolHostDao storagePoolHostDao;
+
+    @Before
+    public void setUp() {
+        ComponentContext.initComponentsLifeCycle();
+    }
 
     @Test(expected = InvalidParameterValueException.class)
     public void testVerifyTemplateIdOfSystemTemplate() {
-        tmgr.verifyTemplateId(1L);
+        templateManager.verifyTemplateId(1L);
     }
 
     public void testVerifyTemplateIdOfNonSystemTemplate() {
-        tmgr.verifyTemplateId(1L);
+        templateManager.verifyTemplateId(1L);
+    }
+
+    @Test
+    public void testPrepareTemplateIsSeeded() {
+        VMTemplateVO mockTemplate = mock(VMTemplateVO.class);
+        when(mockTemplate.getId()).thenReturn(202l);
+
+        StoragePoolVO mockPool = mock(StoragePoolVO.class);
+        when(mockPool.getId()).thenReturn(2l);
+
+        PrimaryDataStore mockPrimaryDataStore = mock(PrimaryDataStore.class);
+        when(mockPrimaryDataStore.getId()).thenReturn(2l);
+
+        VMTemplateStoragePoolVO mockTemplateStore = mock(VMTemplateStoragePoolVO.class);
+        when(mockTemplateStore.getDownloadState()).thenReturn(VMTemplateStorageResourceAssoc.Status.DOWNLOADED);
+
+        when(dataStoreManager.getPrimaryDataStore(anyLong())).thenReturn(mockPrimaryDataStore);
+        when(vmTemplateDao.findById(anyLong(), anyBoolean())).thenReturn(mockTemplate);
+        when(vmTemplatePoolDao.findByPoolTemplate(anyLong(), anyLong())).thenReturn(mockTemplateStore);
+
+        doNothing().when(mockTemplateStore).setMarkedForGC(anyBoolean());
+
+        VMTemplateStoragePoolVO returnObject = templateManager.prepareTemplateForCreate(mockTemplate, (StoragePool) mockPrimaryDataStore);
```

Comment:
You are testing another method then the ones you changed. Why?
The test seems to make sense and please keep it but it doesn't test
prepareTemplate(long templateId, long zoneId, Long storageId)
or
prepareTemplateInOneStoragePool(final VMTemplateVO template, final StoragePoolVO pool)
which are the one you changed and the one you added.